public/viewlog.php

```diff
@@ -44,11 +44,40 @@
 $fDomain = '';
 $error = 0;
 
+
 if ($_SERVER['REQUEST_METHOD'] == "GET") {
-    if ((is_array($list_domains) and sizeof($list_domains) > 0) and !authentication_has_role('global-admin')) {
-        $fDomain = $list_domains[0];
+    if (isset($_GET['page']) && $_GET['fDomain']){
+	$fDomain_aux = escape_string($_GET['fDomain']);		
+	$flag_fDomain = 0;
+	if ((is_array($list_domains) and sizeof($list_domains) > 0)) {
+		foreach ($list_domains as $domain){
+			if ($domain == $fDomain_aux){
+				$fDomain=$domain;
+				$flag_fDomain=1;
+				break;
+			}
+		}
+	}
+	
+	if ($flag_fDomain == 0 ){
+		die('Unknown domain');
+	}	
+
+	$page_number_aux=escape_string($_GET['page']);
```

Comment:
I'd not use 'escape_string' here - i think you'd be best using e.g.

$page_number_aux = (int) ($_GET['page'] ?? 0);

(in more english like ... that's : give me the 'page' url parameter, and if it doesn't exist default it to zero; and cast the entire thing to a number)

Reply:
I modified the line